acceptance/acceptance_test.go

```diff
@@ -222,43 +224,47 @@ func runTest(t *testing.T, dir, coverDir string, repls testdiff.ReplacementsCont
 
 	isEnabled, isPresent := config.GOOS[runtime.GOOS]
 	if isPresent && !isEnabled {
-		t.Skipf("Disabled via GOOS.%s setting in %s", runtime.GOOS, configPath)
+		if !Force {
+			t.Skipf("Disabled via GOOS.%s setting in %s", runtime.GOOS, configPath)
+		}
 	}
 
 	cloudEnv := os.Getenv("CLOUD_ENV")
 	isRunningOnCloud := cloudEnv != ""
 	tailOutput := Tail
 
-	if isRunningOnCloud {
-		if isTruePtr(config.CloudSlow) {
-			if testing.Short() {
-				t.Skipf("Disabled via CloudSlow setting in %s (CLOUD_ENV=%s, Short=%v)", configPath, cloudEnv, testing.Short())
-			}
+	if isRunningOnCloud && isTruePtr(config.CloudSlow) && testing.Verbose() {
+		// Combination of CloudSlow and -v auto-enables -tail
+		tailOutput = true
+	}
 
-			if testing.Verbose() {
-				// Combination of CloudSlow and -v auto-enables -tail
-				tailOutput = true
+	if !Force {
+		if isRunningOnCloud {
+			if isTruePtr(config.CloudSlow) {
+				if testing.Short() {
+					t.Skipf("Disabled via CloudSlow setting in %s (CLOUD_ENV=%s, Short=%v)", configPath, cloudEnv, testing.Short())
+				}
 			}
-		}
 
-		isCloudEnabled := isTruePtr(config.Cloud) || isTruePtr(config.CloudSlow)
-		if !isCloudEnabled {
-			t.Skipf("Disabled via Cloud/CloudSlow setting in %s (CLOUD_ENV=%s, Cloud=%v, CloudSlow=%v)",
-				configPath,
-				cloudEnv,
-				isTruePtr(config.Cloud),
-				isTruePtr(config.CloudSlow),
-			)
-		}
+			isCloudEnabled := isTruePtr(config.Cloud) || isTruePtr(config.CloudSlow)
+			if !isCloudEnabled {
+				t.Skipf("Disabled via Cloud/CloudSlow setting in %s (CLOUD_ENV=%s, Cloud=%v, CloudSlow=%v)",
+					configPath,
+					cloudEnv,
+					isTruePtr(config.Cloud),
+					isTruePtr(config.CloudSlow),
+				)
+			}
 
-		if isTruePtr(config.RequiresUnityCatalog) && os.Getenv("TEST_METASTORE_ID") == "" {
-			t.Skipf("Disabled via RequiresUnityCatalog setting in %s (TEST_METASTORE_ID=%s)", configPath, os.Getenv("TEST_METASTORE_ID"))
-		}
+			if isTruePtr(config.RequiresUnityCatalog) && os.Getenv("TEST_METASTORE_ID") == "" {
+				t.Skipf("Disabled via RequiresUnityCatalog setting in %s (TEST_METASTORE_ID=%s)", configPath, os.Getenv("TEST_METASTORE_ID"))
+			}
 
-	} else {
-		// Local run
-		if !isTruePtr(config.Local) {
-			t.Skipf("Disabled via Local setting in %s (CLOUD_ENV=%s)", configPath, cloudEnv)
+		} else {
+			// Local run
+			if !isTruePtr(config.Local) {
+				t.Skipf("Disabled via Local setting in %s (CLOUD_ENV=%s)", configPath, cloudEnv)
+			}
```

Comment:
The number of conditionals and combinations of them is growing...

Reply:
Yes, there are many things to control the skipping behaviour now: Local, Cloud, CloudSlow, RequiredUnityCatalog, -force, GOOS.

All have very clear and composable semantics (with exception for CloudSlow but that has a reason behind it).

Reply:
I think we need to start maintaining a doc on this, both from the pov of running and how to run acceptance tests, as well as how to write them. I haven't written many (none in the past weeks) and I think it will be copy/paste and trial and error to figure it out.